src/algebra/continued_fractions/convergents_equiv.lean

```diff
@@ -0,0 +1,409 @@
+/-
+Copyright (c) 2020 Kevin Kappelmann. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Kappelmann
+-/
+import algebra.continued_fractions.continuants_recurrence
+import algebra.continued_fractions.terminated_stable
+import tactic.linarith
+/-!
+# Equivalence of Convergent Computations for gcfs
+
+## Summary
+
+We show the equivalence of the convergents computations (recurrence relation (`convergents`) vs.
+direct evaluation (`convergents'`)) for `gcf`s on linear ordered fields. We follow the proof from
+[hardy1979introduction], Chapter 10.
+
+## Main Theorems
+
+- `generalized_continued_fraction.convergents_eq_convergents'` shows the equivalence under a strict
+positivity restriction on the sequence.
+- `continued_fractions.convergents_eq_convergents'` shows the equivalence for (regular) continued
+fractions.
+
+## References
+
+- https://en.wikipedia.org/wiki/Generalized_continued_fraction
+- [Hardy, Godfrey Harold and Wright, Edward Maitland, and others,
+*An introduction to the theory of numbers*][hardy1979introduction]
+
+## Tags
+
+fractions, recurrence, equivalence
+-/
+
+variables {α : Type*} {n : ℕ}
```

Comment:
Not a big deal, but I think we now try to use more mathematically natural names for these types. For instance, division rings and fields are often called `K`.